File: src/examples/cpp/example1.cpp
#include <SDL.h>
#include "../../object.h"
#include "../../init.h"
#include "../../main_cycle.h"
#include "../../object_scene.h"

int main()
{
set_options(640, 480, 32, true, false);
set_caption("Example1", "Test Example 1");
object *first_object = new object_scene;
set_head_object(first_object);
SDL_Thread *cyle_thread = init_cycle();
SDL_WaitThread(cyle_thread, NULL);
return 0;
}

File: src/examples/python/example1.py
import pyscovil, time
pyscovil.set_options(640, 480, 32, True, False)
pyscovil.set_caption("PyTest", "Python Test")

scene_object = pyscovil.object_scene()

pyscovil.set_head_object(scene_object)

cycle = pyscovil.init_cycle()

dot1_change = pyscovil.object_change_dot()
dot1 = dot1_change.body_object_dot
dot1_change.set_size(16)
dot1_change.set_color(pyscovil.vec4(1.0, 0, 0, 1.0));
dot1_change.set_cord(pyscovil.vec3(0, 0, -0.1));
dot1_change.set_upper_object(scene_object)
pyscovil.add_change(dot1_change)

time.sleep(1)

dot1_change = pyscovil.object_change_dot(dot1)
dot1_change.set_size(15)
dot1_change.set_color(pyscovil.vec4(1.0, 1.0, 0, 1.0));
dot1_change.set_cord(pyscovil.vec3(0, 0, -0.1));
pyscovil.add_change(dot1_change)

time.sleep(2)

#pyscovil.SDL_WaitThread(cycle, pyscovil.get_null())
